packages/client/src/components/pages/muster-page/table-unit-muster-compliance.tsx

```diff
@@ -155,6 +119,57 @@ export const UnitMusterComplianceTable = (props: any) => {
   // Functions
   //
 
+  const getUnitComplianceData = async (organizationId: number, unitName: string, isoStartDate: string, isoEndDate: string): Promise<ComplianceData> => {
+    const xRange: string[] = [];
+    const yRange: number[] = [];
+    const colors: string[] = [];
+    let dayCount: number = 0;
+    let totalCompliance: number = 0;
+    const complianceStats: ComplianceStats = {
+      minCompliance: 100.0,
+      maxCompliance: 0,
+      avgCompliance: 0,
+    };
+    const res = await MusterClient.getMusterComplianceByDateRange(organizationId, unitName, { isoStartDate, isoEndDate });
+    if (res.musterComplianceRates.length > 0) {
+      res.musterComplianceRates.forEach(rate => {
+        const compliancePerc = rate.musterComplianceRate * 100;
+        xRange.push(rate.isoDate);
+        if (rate.musterComplianceRate !== null) {
+          yRange.push(compliancePerc);
+          colors.push('#7776AF');
+          // update min compliance
+          if (compliancePerc < complianceStats.minCompliance) {
+            complianceStats.minCompliance = compliancePerc;
+          }
```

Comment:
I wonder if the logic for this might be a little clearer/safer if we did something like this (outside of the loop)...

```typescript
      complianceStats.minCompliance = _.min(
        res.musterComplianceRates.map(x => x.musterComplianceRate * 100),
      );
```

After reading the code for a minute it was obvious what you were doing, but for a moment I was confused when I saw `minCompliance` defined higher than `maxCompliance` above. This way I'm suggesting is obviously less efficient since there would be multiple loops, but I doubt it would matter.

Reply:
Cool I think it makes sense to leave as is to prevent unnecessary iteration